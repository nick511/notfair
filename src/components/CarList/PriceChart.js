import React from 'react'
import PropTypes from 'prop-types'

import styles from './PriceChart.module.scss'

function PriceChart({ realPrice, minPrice, listPrice, fairMaxPrice }) {
  const getPricePercentage = price => {
    let pcercent = ((price - minPrice) / (listPrice - minPrice)) * 100
    return Math.min(Math.max(parseInt(pcercent), 0), 100) // limit number between 0-100
  }

  const formatCurrency = number => '$' + parseInt(number, 10).toLocaleString()

  const realPricePercent = getPricePercentage(realPrice)
  const fairMaxPercent = getPricePercentage(fairMaxPrice)

  return (
    <div className={styles.component}>
      <div className={styles.minPrice}>
        <div className={styles.label}>Min Price</div>
        <div className={styles.price}>{formatCurrency(minPrice)}</div>
      </div>

      <div className={styles.priceBar}>
        <div className={styles.barBox}>
          <div className={styles.bar}>
            <div
              className={styles.colorBar}
              style={{ width: `${fairMaxPercent}%` }}
            />
          </div>
          <div
            className={styles.realPrice}
            style={{ left: `${realPricePercent}%` }}
          >
            <div className={styles.price}>{formatCurrency(realPrice)}</div>
            <div className={styles.line} />
            <div className={styles.roundButton} />
          </div>
          <div
            className={styles.fairMax}
            style={{ left: `${fairMaxPercent}%` }}
          >
            <div className={styles.line} />
            <div className={styles.offset}>
              <div className={styles.label}>Fair Program Max</div>
              <div className={styles.price}>{formatCurrency(fairMaxPrice)}</div>
            </div>
          </div>
        </div>
      </div>

      <div className={styles.listPrice}>
        <div className={styles.label}>List Price</div>
        <div className={styles.price}>{formatCurrency(listPrice)}</div>
      </div>
    </div>
  )
}

PriceChart.propTypes = {
  realPrice: PropTypes.number.isRequired,
  minPrice: PropTypes.number.isRequired,
  listPrice: PropTypes.number.isRequired,
  fairMaxPrice: PropTypes.number.isRequired,
}

export default PriceChart
