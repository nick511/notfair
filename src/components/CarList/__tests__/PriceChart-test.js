import React from 'react'
import { shallow } from 'enzyme'

import PriceChart from '../PriceChart'

describe('<PriceChart />', () => {
  let wrapper = null
  const props = {
    realPrice: 16999,
    minPrice: 12680,
    listPrice: 17999,
    fairMaxPrice: 17750,
  }

  beforeEach(() => {
    wrapper = shallow(<PriceChart {...props} />)
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders prices', () => {
    expect(wrapper.find('.minPrice .price').text()).toEqual(
      '$' + props.minPrice.toLocaleString(),
    )
    expect(wrapper.find('.realPrice .price').text()).toEqual(
      '$' + props.realPrice.toLocaleString(),
    )
  })

  it('produces correct percentage', () => {
    const { minPrice, listPrice, fairMaxPrice } = props
    const fairMaxPercent =
      ((fairMaxPrice - minPrice) / (listPrice - minPrice)) * 100
    expect(wrapper.find('.fairMax').prop('style').left).toEqual(
      fairMaxPercent + '%',
    )
  })
})
