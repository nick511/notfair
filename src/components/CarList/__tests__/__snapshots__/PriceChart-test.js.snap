// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceChart /> renders without crashing 1`] = `
<div
  className="component"
>
  <div
    className="minPrice"
  >
    <div
      className="label"
    >
      Min Price
    </div>
    <div
      className="price"
    >
      $12,680
    </div>
  </div>
  <div
    className="priceBar"
  >
    <div
      className="barBox"
    >
      <div
        className="bar"
      >
        <div
          className="colorBar"
          style={
            Object {
              "width": "95%",
            }
          }
        />
      </div>
      <div
        className="realPrice"
        style={
          Object {
            "left": "81%",
          }
        }
      >
        <div
          className="price"
        >
          $16,999
        </div>
        <div
          className="line"
        />
        <div
          className="roundButton"
        />
      </div>
      <div
        className="fairMax"
        style={
          Object {
            "left": "95%",
          }
        }
      >
        <div
          className="line"
        />
        <div
          className="offset"
        >
          <div
            className="label"
          >
            Fair Program Max
          </div>
          <div
            className="price"
          >
            $17,750
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="listPrice"
  >
    <div
      className="label"
    >
      List Price
    </div>
    <div
      className="price"
    >
      $17,999
    </div>
  </div>
</div>
`;
